src/test/suite/tokenGraphemeSplitter.test.ts

```diff
@@ -14,280 +14,265 @@ import { FakeIDE } from "./fakes/ide/FakeIDE";
  */
 type CompactGrapheme = [string, number, number];
 
-interface TestCase {
-  input: string;
-  expectedOutput: CompactGrapheme[];
-}
+/**
+ * A compact representation of a test case. Expected to be of the form
+ * [input, expectedOutput]
+ */
+type TestCase = [string, CompactGrapheme[]];
 
 interface SplittingModeTestCases {
   tokenHatSplittingMode: Partial<TokenHatSplittingMode>;
   extraTestCases: TestCase[];
 }
 
 const commonTestCases: TestCase[] = [
-  {
-    input: "hi",
-    expectedOutput: [
+  [
+    "hi",
+    [
       ["h", 0, 1],
       ["i", 1, 2],
     ],
-  },
-  {
-    input: "_",
-    expectedOutput: [["_", 0, 1]],
-  },
-  {
-    input: "😄",
-    expectedOutput: [[UNKNOWN, 0, 2]],
-  },
+  ],
+  ["_", [["_", 0, 1]]],
+  ["😄", [[UNKNOWN, 0, 2]]],
 ];
 
 const tests: SplittingModeTestCases[] = [
   {
-    tokenHatSplittingMode: {
-      preserveAccents: true,
-    },
-    extraTestCases: [
-      {
-        input: "\u00F1", // ñ as single codepoint
-        expectedOutput: [["\u00F1", 0, 1]],
-      },
-      {
-        input: "\u006E\u0303", // ñ using combining mark
-        expectedOutput: [["\u00F1", 0, 2]],
-      },
-      {
-        input: "\u00D1", // Ñ as single codepoint
-        expectedOutput: [["\u00F1", 0, 1]],
-      },
-      {
-        input: "\u004E\u0303", // Ñ using combining mark
-        expectedOutput: [["\u00F1", 0, 2]],
-      },
-      {
-        input: "Hi",
-        expectedOutput: [
-          ["h", 0, 1],
-          ["i", 1, 2],
-        ],
-      },
-    ],
-  },
-  {
-    tokenHatSplittingMode: {
-      preserveCase: true,
-      preserveAccents: true,
-    },
+    tokenHatSplittingMode: {},
     extraTestCases: [
-      {
-        input: "\u00F1", // ñ as single codepoint
-        expectedOutput: [["\u00F1", 0, 1]],
-      },
-      {
-        input: "\u006E\u0303", // ñ using combining mark
-        expectedOutput: [["\u00F1", 0, 2]],
-      },
-      {
-        input: "\u00D1", // Ñ as single codepoint
-        expectedOutput: [["\u00D1", 0, 1]],
-      },
-      {
-        input: "\u004E\u0303", // Ñ using combining mark
-        expectedOutput: [["\u00D1", 0, 2]],
-      },
-      {
-        input: "Hi",
-        expectedOutput: [
-          ["H", 0, 1],
-          ["i", 1, 2],
-        ],
-      },
+      [
+        "\u00F1", // ñ as single codepoint
+        [["n", 0, 1]],
+      ],
+      [
+        "\u006E\u0303", // ñ using combining mark
+        [["n", 0, 2]],
+      ],
+      [
+        "\u00D1", // Ñ as single codepoint
+        [["n", 0, 1]],
+      ],
+      [
+        "\u004E\u0303", // Ñ using combining mark
+        [["n", 0, 2]],
+      ],
+      ["ꝏ", [[UNKNOWN, 0, 1]]],
+      ["ø", [["o", 0, 1]]],
+      ["æ", [[UNKNOWN, 0, 1]]],
```

Comment:
I've been playing with these a bit locally while editing the test cases.  It feels pretty good, though part of me wants to say `"air"` for `æ` and `"special"` for `ø`, whereas this PR makes them `"special"` and `"odd"`, respectively.

Reply:
Yeah it is a bit in inconsistent how these are treated. These are proper characters in the danish/Norwegian alphabet
danish/norwegian | swedish
| - | - |
ø | ö
æ | ä